yarn-project/archiver/src/archiver/kv_archiver_store/log_store.ts

```diff
@@ -52,8 +52,13 @@ export class LogStore {
   addLogs(blocks: L2Block[]): Promise<boolean> {
     return this.db.transaction(() => {
       blocks.forEach(block => {
+        const dataStartIndexForBlock =
+          block.header.state.partial.noteHashTree.nextAvailableLeafIndex -
+          block.body.numberOfTxsIncludingPadded * MAX_NOTE_HASHES_PER_TX;
         void this.#noteEncryptedLogsByBlock.set(block.number, block.body.noteEncryptedLogs.toBuffer());
-        block.body.noteEncryptedLogs.txLogs.forEach(txLogs => {
+        block.body.noteEncryptedLogs.txLogs.forEach((txLogs, txIndex) => {
+          const txHash = block.body.txEffects[txIndex].txHash;
+          const dataStartIndexForTx = dataStartIndexForBlock + txIndex * MAX_NOTE_HASHES_PER_TX;
```

Comment:
It'd be good to explain what this value is and why we're storing it

Reply:
Clarified in `TxScopedEncrypteedL2NoteLog`